ProcessLib/SourceTerms/CreateSourceTerm.cpp

```diff
@@ -29,6 +29,35 @@ std::unique_ptr<SourceTerm> createSourceTerm(
     //! \ogs_file_param{prj__process_variables__process_variable__source_terms__source_term__type}
     auto const type = config.config.peekConfigParameter<std::string>("type");
 
+    // check basic data consistency
+    if (variable_id >=
+            static_cast<int>(dof_table_bulk.getNumberOfVariables()) ||
+        *config.component_id >=
+            dof_table_bulk.getNumberOfVariableComponents(variable_id))
+    {
+        OGS_FATAL(
+            "Variable id or component id too high. Actual values: (%d, "
+            "%d), maximum values: (%d, %d).",
```

Comment:
Btw.: the "maximum" is off by one. What is printed is in C style one past maximum. This is confusing. It's wrong everywhere, and I think I made it wrong for the first time. :smile:

Reply:
In `dof_table_bulk.getNumberOfVariables()` the `size - 1` is returned, so the message for the number of variables should be consistent. `dof_table_bulk.getNumberOfVariableComponents(variable_id)` returns a difference where the off by one should be canceled out.